tests/testUtility.cpp

```diff
@@ -0,0 +1,89 @@
+#include "gtest/gtest.h"
+#include <libgen.h>
+
+extern "C" {
+#include "utility.h"
+}
+
+struct UtilityTest : testing::Test {
+    char *modulePath = "";
+
+    UtilityTest() { modulePath = getModulePath(); }
+
+    ~UtilityTest() {}
+};
+
+TEST_F(UtilityTest, getModulePath) {
+    std::string moduleNameFromPath(basename(modulePath));
+
+    std::size_t found = moduleNameFromPath.find("hexrotpxicom");
+    EXPECT_TRUE(found != std::string::npos);
+}
+
+TEST_F(UtilityTest, isFile) {
+    const char *pFilePath = joinStr(modulePath, "/tests/main.cpp");
+    EXPECT_TRUE(isFile(pFilePath));
+}
+
+TEST_F(UtilityTest, isFileWrong) { EXPECT_FALSE(isFile(modulePath)); }
+
+TEST_F(UtilityTest, isDir) { EXPECT_TRUE(isDir(modulePath)); }
+
+TEST_F(UtilityTest, isDirWrong) {
+    const char *pFilePath = joinStr(modulePath, "/tests/main.cpp");
+
+    EXPECT_FALSE(isDir(pFilePath));
+}
+
+TEST(utility, isAbsPath) {
+    const char *absPath = "/opt/home";
+
+    bool isAbs = isAbsPath(absPath);
+    EXPECT_TRUE(isAbs);
+}
+
+TEST(utility, isAbsPathWrong) {
+    const char *absPath = "opt/home";
+
+    bool isAbs = isAbsPath(absPath);
+    EXPECT_FALSE(isAbs);
+}
+
+TEST(utility, joinStr) {
+    char *str1 = "abc";
+    char *str2 = "def";
+    const char *strJoin = joinStr(str1, str2);
+
+    const char *ans = "abcdef";
+    EXPECT_STREQ(ans, strJoin);
+}
+
+TEST(utility, isLimitSwitchOn) {
```

Comment:
This seems specific to the rotator. Can you please add some comments explaining how that works here. At least say that these mask values come from the MT Rotator. Or are just made up, if that's the case.

Reply:
This is for the rotator and hexapod. The hardware engineer did the same modification for the rotator, camera hexapod, and M2 hexapod. Thanks!